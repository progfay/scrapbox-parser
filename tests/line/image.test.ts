/* global describe it expect */

describe('image', () => {
  it('Simple image', () => {
    expect(`[http://example.com/image.png]
[https://example.com/image.JPG]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'http://example.com/image.png',
            link: ''
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://example.com/image.JPG',
            link: ''
          }
        ]
      }
    ])
  })

  it('HTTP jpeg image with special and japanese chars', () => {
    expect('[http://example.com/~!@#$%^&*()_+`-={}\\\'"?,.<>|/画像.jpeg]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'http://example.com/~!@#$%^&*()_+`-={}\\\'"?,.<>|/画像.jpeg',
            link: ''
          }
        ]
      }
    ])
  })

  it('HTTPS svg and GIF image with link', () => {
    expect(`[https://example.com/image.svg https://example.com/]
[https://example.com/ https://example.com/image.GIF]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://example.com/image.svg',
            link: 'https://example.com/'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://example.com/image.GIF',
            link: 'https://example.com/'
          }
        ]
      }
    ])
  })

  it('Image with double image link', () => {
    expect('[https://example.com/forward.png https://example.com/backward.png]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://example.com/backward.png',
            link: 'https://example.com/forward.png'
          }
        ]
      }
    ])
  })

  it('Gyazo image', () => {
    expect(`[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]
[https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815]
[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/raw]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000',
            link: ''
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000',
            link: ''
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/raw',
            link: ''
          }
        ]
      }
    ])
  })

  it('Gyazo image with link', () => {
    expect(`[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815 https://example.com]
[https://example.com https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]
[https://gyazo.com/7057219f5b20ca8afd122945b72453d3 https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]`).toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000',
            link: 'https://example.com'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000',
            link: 'https://example.com'
          }
        ]
      },
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000',
            link: 'https://gyazo.com/7057219f5b20ca8afd122945b72453d3'
          }
        ]
      }
    ])
  })

  it('Image with GET parameters', () => {
    expect('[http://example.com/image.png?key1=value1&key2=value2]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'image',
            src: 'http://example.com/image.png?key1=value1&key2=value2',
            link: ''
          }
        ]
      }
    ])
  })
})
