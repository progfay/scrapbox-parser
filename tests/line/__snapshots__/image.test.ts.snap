// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`image Direct Gyazo image 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815.png]",
        "src": "https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815.png",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image Gyazo image 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]",
        "src": "https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000",
        "type": "image",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815]",
        "src": "https://i.gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000",
        "type": "image",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/raw]",
        "src": "https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/raw",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image Gyazo image with link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "https://example.com",
        "raw": "[https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815 https://example.com]",
        "src": "https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000",
        "type": "image",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "https://example.com",
        "raw": "[https://example.com https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]",
        "src": "https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000",
        "type": "image",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "https://gyazo.com/7057219f5b20ca8afd122945b72453d3",
        "raw": "[https://gyazo.com/7057219f5b20ca8afd122945b72453d3 https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815]",
        "src": "https://gyazo.com/0f82099330f378fe4917a1b4a5fe8815/thumb/1000",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image HTTP jpeg image with special and japanese chars 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[http://example.com/~!@#$%^&*()_+\`-={}\\\\'\\"?,.<>|/画像.jpeg]",
        "src": "http://example.com/~!@#$%^&*()_+\`-={}\\\\'\\"?,.<>|/画像.jpeg",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image HTTPS svg and GIF image with link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "https://example.com/",
        "raw": "[https://example.com/image.svg https://example.com/]",
        "src": "https://example.com/image.svg",
        "type": "image",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "https://example.com/",
        "raw": "[https://example.com/ https://example.com/image.GIF]",
        "src": "https://example.com/image.GIF",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image Image with GET parameters 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[http://example.com/image.png?key1=value1&key2=value2]",
        "src": "http://example.com/image.png?key1=value1&key2=value2",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image Image with double image link 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "https://example.com/forward.png",
        "raw": "[https://example.com/forward.png https://example.com/backward.png]",
        "src": "https://example.com/backward.png",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;

exports[`image Simple image 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[http://example.com/image.png]",
        "src": "http://example.com/image.png",
        "type": "image",
      },
    ],
    "type": "line",
  },
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "link": "",
        "raw": "[https://example.com/image.JPG]",
        "src": "https://example.com/image.JPG",
        "type": "image",
      },
    ],
    "type": "line",
  },
]
`;
